lib/child_process.js

```diff
@@ -841,17 +862,16 @@ function checkExecSyncError(ret, args, cmd) {
  *   }} [options]
  * @returns {Buffer | string}
  */
-function execFileSync(command, args, options) {
-  options = normalizeSpawnArguments(command, args, options);
+function execFileSync(file, args, options) {
+  ({ file, args, options } = normalizeExecFileArgs(file, args, options));
 
   const inheritStderr = !options.stdio;
-  const ret = spawnSync(options.file,
-                        ArrayPrototypeSlice(options.args, 1), options);
+  const ret = spawnSync(file, args, options);
 
   if (inheritStderr && ret.stderr)
     process.stderr.write(ret.stderr);
 
-  const err = checkExecSyncError(ret, options.args, undefined);
+  const err = checkExecSyncError(ret, [options.argv0 || file, ...args]);
```

Comment:
Array spreading relies on globally mutable `Array.prototype[Symbol.iterator]`, can we avoid that?
```suggestion
  const errArgs = [options.argv0 || file]
  ArrayPrototypePushApply(errArgs, args);
  const err = checkExecSyncError(ret, errArgs);
```
Alternatively, we could change the signature of `checkExecSyncError` since it's an internal function,

Reply:
Array spreading using `...` is banned on JS layer of Node.js?  How about Object spreading?

Reply:
Can‘t change the function signature of `checkExecSyncError()`, because this function is not used only here. I replaced spread with unshift

Reply:
> To clarify, the spread operator is fine, using it on an array is not.
> 
> https://github.com/nodejs/node/blob/df5664440e447c2724003700b5a81da8fadc4f5a/doc/contributing/primordials.md#L251-L266
> 
> Object spreading does not rely on any mutable method AFAIK, so it's perfectly fine to use it in core.

Thank you for your information. I learn something.

Reply:
@zhmushan `ArrayPrototypePushApply`'s second argument is an array, so there is no need to use `...args`.


`primordials.md` has an example about how to use `ArrayPrototypePushApply` to achieve array concation.

https://github.com/nodejs/node/blob/595ce9dac63cdd32d893c3319328a752719644d2/doc/contributing/primordials.md?plain=1#L306-L320

`Function.prototype.apply()` at mdn: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

Reply:
@F3n67u Thanks! Your elaboration was very helpful to me and I have corrected my code.

Reply:
Unshift would mutate the `args` array, which may be problematic (or maybe not, I haven't tried it), but would certainly be fine otherwise.